runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp

```diff
@@ -0,0 +1,1536 @@
+					bsmData += (bsmData[1] + 2);
+				}
+
+				U_16 bsmCPIndex = bsmData[0];
+				U_16 argCount = bsmData[1];
+				bsmData += 2;
+
+				j9object_t obj;
+				j9object_t ifNotAvailableObject = NULL;
+				if (NULL != ifNotAvailable) {
+					ifNotAvailableObject = J9_JNI_UNWRAP_REFERENCE(ifNotAvailable);
+				}
+				while (start < end) {
+					/* Copy the arguments between start and end to the buf array
+					 *
+					 * Negative start index refer to the mandatory arguments for a bootstrap method
+					 * -4 -> Lookup
+					 * -3 -> name (String)
+					 * -2 -> signature (MethodType)
+					 * -1 -> argCount of optional arguments
+					 */
+					obj = NULL;
+					if (start >= 0) {
+						U_16 argIndex = bsmData[start];
+						J9ConstantPool *ramConstantPool = J9_CP_FROM_CLASS(callerClass);
+						obj = resolveRefToObject(currentThread, ramConstantPool, argIndex, (JNI_TRUE == resolve));
+						if ((NULL == obj) && (JNI_TRUE != resolve)) {
+							obj = ifNotAvailableObject;
+						}
+					} else if (start == -4) {
+						obj = resolveRefToObject(currentThread, J9_CP_FROM_CLASS(callerClass), bsmCPIndex, true);
+					} else if (start == -3) {
+						J9UTF8 *name = J9ROMNAMEANDSIGNATURE_NAME(nameAndSig);
+						obj = vm->memoryManagerFunctions->j9gc_createJavaLangString(currentThread, J9UTF8_DATA(name), (U_32)J9UTF8_LENGTH(name), J9_STR_INTERN);
+					} else if (start == -2) {
+						J9UTF8 *signature = J9ROMNAMEANDSIGNATURE_SIGNATURE(nameAndSig);
+						/* Call VM Entry point to create the MethodType - Result is put into the
+						* currentThread->returnValue as entry points don't "return" in the expected way
+						*/
+						vmFuncs->sendFromMethodDescriptorString(currentThread, signature, callerClass->classLoader, NULL);
+						obj = (j9object_t)currentThread->returnValue;
+					} else if (start == -1) {
+						obj = vm->memoryManagerFunctions->J9AllocateObject(currentThread, J9VMJAVALANGINTEGER_OR_NULL(vm), J9_GC_ALLOCATE_OBJECT_NON_INSTRUMENTABLE);
+						J9VMJAVALANGINTEGER_SET_VALUE(currentThread, obj, argCount);
+					}
+					J9JAVAARRAYOFOBJECT_STORE(currentThread, bufferArray, pos, obj);
+					start += 1;
+					pos += 1;
+				}
+			} else {
+				vmFuncs->setCurrentExceptionUTF(currentThread, J9VMCONSTANTPOOL_JAVALANGLINKAGEERROR, NULL);
+			}
+		}
+	}
+	vmFuncs->internalExitVMToJNI(currentThread);
+}
+
+/**
+ * private static native void clearCallSiteContext(CallSiteContext context);
+ */
+void JNICALL
+Java_java_lang_invoke_MethodHandleNatives_clearCallSiteContext(JNIEnv *env, jclass clazz, jobject context)
+{
+	return;
+}
+
+/**
+ * private static native int getNamedCon(int which, Object[] name);
+ */
+jint JNICALL
+Java_java_lang_invoke_MethodHandleNatives_getNamedCon(JNIEnv *env, jclass clazz, jint which, jobjectArray name)
+{
+	return 0;
+}
+
+/**
+ * private static native void registerNatives();
+ */
+void JNICALL
+Java_java_lang_invoke_MethodHandleNatives_registerNatives(JNIEnv *env, jclass clazz)
+{
+	return;
```

Comment:
Why export the functions and make all the prototype changes to different headers rather than using the `registerNatives` function?  Given we're going to have to call it, using it would seem worthwhile....